### DISCUSSION TITLE
How to make a binding to go directly to log_current?

### DISCUSSION THREAD

**Discussion by juliekoubova** [NeogitOrg/neogit#1002] [Q&A]
Hello, how do I make a binding to go directly to current log? I'm trying this, but it doesn't do anything.
```lua
  vim.keymap.set('n', '<leader>gl', function()
    require('neogit').action('log', 'log_current')
  end)
```

**Comment by CKolkey** [comment]
`require('neogit').action('log', 'log_current')` returns a function, so the mapping would look like:

```lua
 vim.keymap.set('n', '<leader>gl', require('neogit').action('log', 'log_current'))
```

But also, that one won't _quite_ work. The log popup has some internal arguments. I'll try to fix that.

**Reply by juliekoubova** [reply]
heh, yeah I thought so, but it didn't work when I tried calling it, so I figured
it must be something else. thank you :)

**Reply by CKolkey** [reply]
Yeah, it's because `--graph` and `--color` never make it to the CLI. I handle that in a parallel call and collate it before rendering. The log with a graph is a proper pain to parse.

**Reply by juliekoubova** [reply]
I haven't seen the code, so maybe I'm talking out of my ass, but wouldn't it be easier to go from commit to commit and render the graph yourself, instead of parsing `git log --graph` output?

**Reply by juliekoubova** [reply]
also, on a completely unrelated note, I am seeing a lot of this. it doesn't seem to break anything, but it is noisy
```
> git --no-pager --literal-pathspecs --no-optional-locks -c core.preloadindex=true -c color.ui=always 
describe --long --tags HEAD
fatal: No tags can describe '615670aa4b565b4abff0c474923d153dad4a24bb'.
Try --always, or create some tags.
```

**Reply by CKolkey** [reply]
Yeah, thats fair. And actually, you can do both - `graph_style` defaults to `"ascii"`, which is parsed output from the CLI, but can be set to `"unicode"` to use the renderer from vim-flog.

I guess in terms of what I know how to do, parsing the graph was easier than writing a algorithm to generate it myself. [Collating it pretty simple](https://github.com/NeogitOrg/neogit/blob/master/lua/neogit/lib/git/log.lua#L191-L210), [the commits can be parsed as JSON](https://github.com/NeogitOrg/neogit/blob/master/lua/neogit/lib/git/log.lua#L311-L378), and and [ANSI colors can be parsed with lua patterns](https://github.com/NeogitOrg/neogit/blob/master/lua/neogit/lib/ansi.lua).

[Constructing a graph, however, is voodoo.](https://github.com/NeogitOrg/neogit/blob/master/lua/neogit/lib/graph.lua).

**Reply by CKolkey** [reply]
Oh, and parsing the graph itself is only `git log --graph --format='%x1E%H%x00'`, not the full log. Two different CLI calls.

**Reply by juliekoubova** [reply]
yay the log action works now! thank you!!!

**Comment by CKolkey** [comment]
> also, on a completely unrelated note, I am seeing a lot of this. it doesn't seem to break anything, but it is noisy
> 
> ```
> > git --no-pager --literal-pathspecs --no-optional-locks -c core.preloadindex=true -c color.ui=always 
> describe --long --tags HEAD
> fatal: No tags can describe '615670aa4b565b4abff0c474923d153dad4a24bb'.
> Try --always, or create some tags.
> ```

Oh, that's annoying. I thought I had that fail more gracefully. I'll fix it, thanks.

**Reply by CKolkey** [reply]
I can't seem to reproduce this in a repo with no tags (and running it as a normal CLI call yields that error). You're on the latest build?

**Reply by juliekoubova** [reply]
I just updated to the commit made in the last 10 minutes or so. It still reproes. There is something wrong with the repo:
```
> git describe HEAD
fatal: No tags can describe '2ec3a8afb26e29449932b68e46e2dfffb6b5739c'.
Try --always, or create some tags.

> git describe --always HEAD
2ec3a8afb

> git --version
git version 2.43.0.windows.1
```
